cairo/ethereum/utils/numeric.cairo

```diff
@@ -165,3 +166,34 @@ func U256_from_be_bytes20{range_check_ptr, bitwise_ptr: BitwiseBuiltin*}(bytes20
     tempvar res = U256(new U256Struct(low=low, high=high));
     return res;
 }
+
+// @dev Panics if overflow
+func U256_add{range_check_ptr}(a: U256, b: U256) -> U256 {
+    alloc_locals;
+    let (res, carry) = uint256_add([a.value], [b.value]);
+
+    with_attr error_message("OverflowError") {
+        assert carry = 0;
+    }
+
+    tempvar result = U256(new U256Struct(res.low, res.high));
+    return result;
+}
+
+// @dev Panics if underflow with OverflowError
+func U256_sub{range_check_ptr}(a: U256, b: U256) -> U256 {
+    alloc_locals;
+    let is_within_bounds = U256_le(b, a);
+    with_attr error_message("OverflowError") {
+        assert is_within_bounds.value = 1;
+    }
+    let (result) = uint256_sub([a.value], [b.value]);
+    tempvar res = U256(new U256Struct(result.low, result.high));
+    return res;
+}
+
+func U256_le{range_check_ptr}(a: U256, b: U256) -> bool {
+    let (result) = uint256_le([a.value], [b.value]);
+    tempvar res = bool(result);
+    return res;
+}
```

Comment:
nit: could be inlined ?
it seems not necessary to put in a function

Reply:
inlined where? in all the places we'll use it?